lib/bulk_data/cache.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module BulkData
+  class Cache
+    include Singleton
```

Comment:
Nice! I'm surprised you don't need to `require "singleton"` here.

Reply:
Ah Rails auto/eager loading takes care of it.